woocommerce/woocommerce-android
# [HACKWeek] Slowness of the reviews fetching

Closes: #6178


### Description

We run 2 calls in parallel, and until the last one is finished, we don't show the data unless it is already in the memory cache. The real root cause is presumably (I didn't do any research on that) that the backend, instead of running 1 query passing all ids in it, makes multiple queries when we pass many ids in there. To mitigate the slowness, I change the logic on our side and just update the screen from the cache whatever the notification call or reviews call comes

```
?number=200&num_note_items=20&fields=id%2Ctype%2Cread%2Cbody%2Csubject%2Ctimestamp%2Cmeta%2Cnote_hash&ids=8022887069%2C%208017108377%2C%208016343286%2C%208011748262%2C%208005976953%2C%208001196867&locale=en_US
```

![image](https://github.com/woocommerce/woocommerce-android/assets/4923871/12b0d7ff-00c9-47b8-b5f4-82d78630fe9e)


### Testing instructions

Try look upon reviews both via:
* More -> reviews
* Products -> Product -> Reviews

Notice that no regression there and that somewhat faster, although still quite slow in some cases

### Images/gif
